### ISSUE TITLE
Error using build-in neovim diagnostics

### ISSUE THREAD

**Issue by ner0-m** [wbthomason/lsp-status.nvim#19]
As [neovim/neovim#12655](https://github.com/neovim/neovim/pull/12655) was merged and using neovim with it, I get the following:

> Error detected while processing function LspStatus:
> line    2:
> buf_diagnostics_count is deprecated. Use 'vim.lsp.diagnostic.get_count'

With my function to update the statusline being:

```lua
function! LspStatus() abort
  if luaeval('#vim.lsp.buf_get_clients() > 0')
    return luaeval("require('lsp-status').status()")
  endif
  return ''
endfunction
```

The error seems to be in `lsp-status/diagnostics.lua:11` Changing it to:

```lua
result[k] = vim.lsp.diagnostic.get_count(level)
```

fixes my error.

**Comment by rockerBOO** [comment]
#18 should resolve this

**Comment by wbthomason** [comment]
Thanks @babariviere for the quick PR! (and @tjdevries for the quick merge)

 @ner0-m, does this resolve your problem?

**Comment by creativenull** [comment]
Hey there! I'm not OP but I am also getting a similar error. That is, after updating the plugin with the included merge #18

```
Error executing lua ...lsp-status.nvim/lua/lsp-status/diagnostics.lua:12: attempt to index field 'diagnostic' (a nil value)
```

**Comment by babariviere** [comment]
@creativenull You need to upgrade to the latest version of nightly neovim 😄

**Comment by creativenull** [comment]
Oops you're right, I took the easy way out (installed via appimage, 2 days old 😅) and didn't see that it was related to a recently updated nightly.

Let me make a build from source and report back...

**Comment by jasonrhansen** [comment]
@ner0-m, @rockerBOO
 
`vim.lsp.diagnostic.get_count` takes a buffer number before the severity.

Wouldn't it need to be changed to
```lua
result[k] = vim.lsp.diagnostic.get_count(0, level)
```
to get the count for the current buffer?

**Comment by wbthomason** [comment]
@jasonrhansen: Nice catch; you're correct. Addressed in 9750f45.

**Comment by ner0-m** [comment]
@wbthomason Thanks I just tried and it seems to work perfectly fine! Then this can be closed :smile:
